tests/api/db/test_hub.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2018 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from sqlalchemy.orm import Session
+
+import mlrun.api.initial_data
+import mlrun.common.schemas
+from mlrun.api.db.base import DBInterface
+
+
+def test_data_migration_rename_marketplace_kind_to_hub(
+    db: DBInterface, db_session: Session
+):
+    # create hub sources
+    for i in range(3):
+        source_name = f"source-{i}"
+        source = mlrun.common.schemas.IndexedHubSource(
+            source=mlrun.common.schemas.HubSource(
+                metadata=mlrun.common.schemas.HubObjectMetadata(
+                    name=source_name, description="some source"
+                ),
+                spec=mlrun.common.schemas.HubSourceSpec(
+                    path="/local/path/to/source", channel="development"
+                ),
+            )
```

Comment:
Creating an "old" source with current source I think is problematic as you might save "new" stuff as part of the preparation of the old data, I would suggest although it might not be the cleanest to actually store it with a dictionary and not use existing classes of schemas. take a look at `test_data_migration_align_runs_table`

Reply:
`store_hub_source` only accepts an `IndexedHubSource`, not dict (as opposed to `store_run` that does accept a dict).
Also, the test is only testing the kind changes, so only the old kind is needed to be mocked, which is done in the rows below.